e3sm_diags/e3sm_diags_driver.py

```diff
@@ -410,5 +453,75 @@ def main(parameters=[]) -> List[CoreParameter]:  # noqa B006
     return parameters_results
 
 
+def _log_diagnostic_run_info(prov_paths: ProvPaths):
+    """Logs information about the diagnostic run.
+
+    This method is useful for tracking the provenance of the diagnostic run
+    and understanding the context of the diagnostic results.
+
+    It logs the following information:
+        - Timestamp of the run
+        - Version information (Git branch and commit hash or module version)
+        - Paths to the provenance files (log, parameter files, Python script,
+          env yml, index HTML)
+
+    Parameters
+    ----------
+    prov_paths : ProvPaths
+        The paths to the provenance files.
+
+    Notes
+    -----
+    The version information is retrieved from the current Git branch and
+    commit hash. If the Git information is not available, it falls back
+    to the version defined in the `e3sm_diags` module.
+    """
+    timestamp = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+
+    try:
+        branch_name = (
+            subprocess.check_output(
+                ["git", "rev-parse", "--abbrev-ref", "HEAD"],
+                cwd=os.path.dirname(__file__),
+                stderr=subprocess.DEVNULL,
+            )
+            .strip()
+            .decode("utf-8")
+        )
+        commit_hash = (
+            subprocess.check_output(
+                ["git", "rev-parse", "HEAD"],
+                cwd=os.path.dirname(__file__),
+                stderr=subprocess.DEVNULL,
```

Comment:
Suppress `git` command error if a versioned release is installed a conda environment instead.